File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.js
/**
 * @param {number} n
 * @return {number}
 */
var numTrees = function(n) {
    let trees = new Array(n+1).fill(1)

    for(let i=2; i<=n; i++){
        let ans = 0;
        for(let j=1; j<=i; j++){
            ans += trees[j-1]*trees[i-j]
        }
        trees[i] = ans
    }

    return trees[n]
};

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.c
int numTrees(int n) {
 int trees[n+1];
 for(int i=0; i<n+1; i++){
    trees[i] = 1;
 }   
 for(int a=2; a<=n; a++){
    int ans = 0;
    for(int b=1; b<= a; b++){
        ans += trees[b-1]*trees[a-b];
    }
    trees[a] = ans;
 }

 return trees[n];
}